pkg/kapp/resources/resource.go

```diff
@@ -138,6 +140,17 @@ func (r *ResourceImpl) GroupVersionResource() schema.GroupVersionResource {
 	return r.resType.GroupVersionResource
 }
 
+func (r *ResourceImpl) GroupVersionKind() schema.GroupVersionKind {
+	return r.un.GroupVersionKind()
+}
+
+func (r *ResourceImpl) GroupKind() schema.GroupKind {
+	return schema.GroupKind{
+		Group: r.APIGroup(),
+		Kind:  r.Kind(),
+	}
```

Comment:
Maybe use `r.un.GroupVersionKind().GroupKind()`?

Reply:
Yep, definitely better. Adopted this.